x/validators/model.go

```diff
@@ -29,28 +29,36 @@ func (m ValidatorUpdates) Validate() error {
 // Store stores ValidatorUpdates to the KVStore while cleaning up those with 0
 // power.
 func (m ValidatorUpdates) Store(store weave.KVStore) error {
-	cleanValidatorSlice := make([]ValidatorUpdate, 0, len(m.ValidatorUpdates))
-	// Cleanup validators with power 0 as these get discarded by tendermint.
+	duplicates := make(map[string]int, 0)
+	cleanValidatorSlice := make([]weave.ValidatorUpdate, 0, len(m.ValidatorUpdates))
```

Comment:
I think there is a way to unify this logic with StoreApp.AddValChanges. Ideally along with the top level weave.ValidatorUpdates. To make sure we catch the same cases everywhere

Reply:
Yes, I'm thinking about using a helper for that.